### ISSUE TITLE
LDpred2: Character in chromosome column causes error

### ISSUE THREAD

**Issue by deepchocolate** [comorment/containers#138]
**Describe the bug**
If using sumstats that contain contain character in the chromosome column this will cause an error as below. In this case due to the presence of "X".
![image](https://user-images.githubusercontent.com/16337160/220584652-05c3698c-5a24-4762-be0a-daaecd4be028.png)

**Expected behavior**
This is unclear.

**Additional context**
As of now, happens when users join up chromosome and position from HRC data to sumstat files generated by meta-analyses.

**Comment by espenhgn** [comment]
I would've assumed `bigreadr` assumes that the sumstats `chr` column should only be numeric, but couldn't find a clear statement that values in each column all must have the same type. Can `bigreadr::fread2` read the file as on [this](https://privefl.github.io/bigsnpr-extdoc/polygenic-scores-pgs.html) page?

Anyway, perhaps this issue is better opened at https://github.com/privefl/bigsnpr/issues instead? There may be a sort of similar case with `snp_match()` reporte here: https://github.com/privefl/bigsnpr/issues/120

**Comment by deepchocolate** [comment]
> I would've assumed `bigreadr` assumes that the sumstats `chr` column should only be numeric, but couldn't find a clear statement that values in each column all must have the same type. Can `bigreadr::fread2` read the file as on [this](https://privefl.github.io/bigsnpr-extdoc/polygenic-scores-pgs.html) page?
> 
> Anyway, perhaps this issue is better opened at https://github.com/privefl/bigsnpr/issues instead? There may be a sort of similar case with `snp_match()` reporte here: [privefl/bigsnpr#120](https://github.com/privefl/bigsnpr/issues/120)

Yeah, I think if the intended use is to only do numeric matching on chromosome the `snp_match` function should list these rows as excluded. But for our purposes It's probably faster just to filter these rows out

**Comment by espenhgn** [comment]
The `vctrs::vec_in` function may accept `NA` for missing values (https://vctrs.r-lib.org/reference/vec_match.html), so replacing `'X'` by `NA` could also by an option.
